### DISCUSSION TITLE
Quan X使用Spotify插件时能够进行翻译，但不能使用外部源

### DISCUSSION THREAD

**Discussion by ChirsHu** [DualSubs/.github#2] [Q&A 问答] [answered]
版本：Ipad os18.1/ios 18.1
Quanx版本：1.5.0-830
Spotify版本：8.9.86.419
问题描述：在boxjs商店把Spotify的翻译歌词和外部源都打开，之后打开Spotify。发现在Spotify原有歌词上能够进行翻译操作，但Spotify没有歌词的歌曲并没有成功调取外部源。在Spotify没有歌词的歌曲页面，日志仅显示：
2024/11/07 01:32:10:905  <console>    [JS Console]: 2024-06-13
2024/11/07 01:32:10:913  <console>    [JS Console]: $response.status不为200:304

在成功翻译的页面日志显示：
2024/11/07 01:46:14:242  <console>    [JS Console]: 🍿️ DualSubs: 🎵 Spotify
2024/11/07 01:46:14:257  <console>    [JS Console]: request.bundle.js
2024/11/07 01:46:14:260  <console>    [JS Console]: Version: 1.9.5
2024/11/07 01:46:14:264  <console>    [JS Console]: Date: 2024/10/31 20:14:46
2024/11/07 01:46:14:276  <console>    [JS Console]: ⚠ url: https://spclient.wg.spotify.com/color-lyrics/v2/track/3AJwUDP919kvQ9QcozQPxg?clientLanguage=zh
2024/11/07 01:46:14:279  <console>    [JS Console]: ⚠ METHOD: GET, HOST: spclient.wg.spotify.com, PATH: /color-lyrics/v2/track/3AJwUDP919kvQ9QcozQPxg
2024/11/07 01:46:14:284  <console>    [JS Console]: ⚠ FORMAT: application/protobuf
2024/11/07 01:46:14:290  <console>    [JS Console]: ☑️ Set Environment Variables
2024/11/07 01:46:14:304  <console>    [JS Console]: ✅ Set Environment Variables, Settings: object, Settings内容: {"Switch":true,"Type":"Translate","Types":["Translate","External"],"Languages":["AUTO","ZH"],"CacheSize":50,"CountryCode":"US"}
2024/11/07 01:46:14:309  <console>    [JS Console]: ⚠ Settings.Switch: true
2024/11/07 01:46:14:316  <console>    [JS Console]: ⚠ Type: Translate, Languages: AUTO,ZH
2024/11/07 01:46:14:321  <console>    [JS Console]: 🚧 调试信息
trackId: 3AJwUDP919kvQ9QcozQPxg
2024/11/07 01:46:14:692  <console>    [JS Console]: ☑️ Set Cache, cacheSize: 50
2024/11/07 01:46:14:697  <console>    [JS Console]: ✅ Set Cache
2024/11/07 01:46:14:717  <console>    [JS Console]: 🚧 调试信息
$request.url: https://spclient.wg.spotify.com/color-lyrics/v2/track/3AJwUDP919kvQ9QcozQPxg?clientLanguage=zh&subtype=Translate
2024/11/07 01:46:14:722  <console>    [JS Console]: 
🚩 执行结束!
2024/11/07 01:46:15:127  <console>    [JS Console]: 🍿️ DualSubs: 🔣 Universal
2024/11/07 01:46:15:138  <console>    [JS Console]: Translate.response.bundle.js
2024/11/07 01:46:15:142  <console>    [JS Console]: Version: 1.6.2
2024/11/07 01:46:15:147  <console>    [JS Console]: Date: 2024/11/5 14:45:32
2024/11/07 01:46:15:171  <console>    [JS Console]: 
🟧 URL v2.1.2
2024/11/07 01:46:15:179  <console>    [JS Console]: ⚠ url: {"protocol":"https:","host":"spclient.wg.spotify.com","hostname":"spclient.wg.spotify.com","port":"","pathname":"/color-lyrics/v2/track/3AJwUDP919kvQ9QcozQPxg","paths":["color-lyrics","v2","track","3AJwUDP919kvQ9QcozQPxg"],"search":"?clientLanguage=zh&subtype=Translate","searchParams":{},"harf":"https://spclient.wg.spotify.com/color-lyrics/v2/track/3AJwUDP919kvQ9QcozQPxg?clientLanguage=zh&subtype=Translate"}
2024/11/07 01:46:15:184  <console>    [JS Console]: ⚠ PATHs: color-lyrics,v2,track,3AJwUDP919kvQ9QcozQPxg
2024/11/07 01:46:15:188  <console>    [JS Console]: ⚠ FORMAT: application/vnd.google.protobuf
2024/11/07 01:46:15:195  <console>    [JS Console]: ☑️ Detect Platform
2024/11/07 01:46:15:200  <console>    [JS Console]: ✅ Detect Platform, Platform: Spotify
2024/11/07 01:46:15:203  <console>    [JS Console]: ⚠ PLATFORM: Spotify
2024/11/07 01:46:15:208  <console>    [JS Console]: ☑️ Set Environment Variables
2024/11/07 01:46:15:219  <console>    [JS Console]: ✅ Set Environment Variables, Settings: object, Settings内容: {"Switch":true,"Type":"Translate","Types":["Translate","External"],"Languages":["AUTO","ZH"],"CacheSize":10,"Vendor":"Google","ShowOnly":false,"Position":"Forward","Method":"Part","Times":3,"Interval":500,"Exponential":true,"GoogleCloud":{"Version":"v2","Mode":"Key","Auth":""},"Microsoft":{"Version":"Azure","Mode":"Token","Region":"","Auth":""},"DeepL":{"Version":"Free","Auth":""},"DeepLX":{"Endpoint":"","Auth":""},"URL":"","NeteaseMusic":{"PhoneNumber":"","Password":""}}
2024/11/07 01:46:15:223  <console>    [JS Console]: ⚠ Type: Translate, Languages: AUTO,ZH
2024/11/07 01:46:15:233  <console>    [JS Console]: ☑️ Translator, vendor: Google, method: Part, [source, target]: EN,ZH
2024/11/07 01:46:15:237  <console>    [JS Console]: ⚠ Chunk Array
2024/11/07 01:46:15:241  <console>    [JS Console]: ☑️ retry, 剩余重试次数:3
时间间隔:500ms
2024/11/07 01:46:15:246  <console>    [JS Console]: 
🟧 Translate v1.0.6
2024/11/07 01:46:15:973  <console>    [JS Console]: ✅ Translator
2024/11/07 01:46:15:985  <console>    [JS Console]: 
🚩 执行结束!
[iOS-Quantumult X Tunnel 2024-11-07-013156.log](https://github.com/user-attachments/files/17651053/iOS-Quantumult.X.Tunnel.2024-11-07-013156.log)

**Comment by VirgilClyne** [comment, marked as answer]
`外部源` 默认针对的是有`歌词面板`，但`歌词面板`显示`无法载入这首歌曲的歌词`的情况
有`歌词面板`的情况下，`Spotify`才会发送网络请求获取歌词，插件才能发挥作用
没有`歌词面板`的情况下，插件做不了任何事
